test/powershell/Language/Parser/Conversions.Tests.ps1

```diff
@@ -73,3 +73,57 @@ Describe 'conversion syntax' -Tags "CI" {
         }
     }
 }
+
+Describe "Type resolution should prefer assemblies in powershell assembly cache" -Tags "Feature" {
+
+    BeforeAll {
+        $cmdletCode = @'
+namespace TestTypeResolution {
+    using System.Management.Automation;
+    [Cmdlet("Test", "TypeResolution")]
+    public class TestTypeResolutionCommand : PSCmdlet {
+        [Parameter()]
+        public string Name { get; set; }
+
+        protected override void BeginProcessing() {
+            WriteObject(Name);
+        }
+    }
+
+    public class TestTypeFoo {
+        public string Foo { get; set; }
+    }
+}
+'@
+        $dupTypeCode = @'
+namespace TestTypeResolution {
+    public class TestTypeFoo {
+        public string Bar { get; set; }
+    }
+}
+'@
+
+        $cmdletDllDir  = Join-Path $TestDrive "cmdlet"
+        $dupTypeDllDir = Join-Path $TestDrive "dupType"
+
+        $null = New-Item -Path $cmdletDllDir, $dupTypeDllDir -ItemType Directory -Force
+
+        $cmdletDllPath  = Join-Path $cmdletDllDir "TestCmdlet.dll"
+        $dupTypeDllPath = Join-Path $dupTypeDllDir "TestType.dll"
+
+        Add-Type $cmdletCode -OutputAssembly $cmdletDllPath
+        Add-Type $dupTypeCode -OutputAssembly $dupTypeDllPath
+
+        $powershell = Join-Path $PSHOME "powershell"
+    }
+
+    It "validate Type resolution should prefer the assembly loaded by Import-Module" {
+        $command = @"
+            Add-Type -Path $dupTypeDllPath
+            Import-Module $cmdletDllPath
+            [TestTypeResolution.TestTypeFoo].Assembly.Location

```

Comment:
That's the existing behavior that I didn't change in this PR, please take a look at the comments [here](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/parser/TypeResolver.cs#L264).
It would be a breaking change if we start to report ambiguity when searching types from all loaded assemblies.